File: lib_base/src/main/java/com/czl/lib_base/mvvm/viewmodel/ItemViewModel.java
package com.czl.lib_base.mvvm.viewmodel;

import androidx.annotation.NonNull;

import com.czl.lib_base.base.BaseViewModel;

public class ItemViewModel<VM extends BaseViewModel> {
    protected VM viewModel;

    public ItemViewModel(@NonNull VM viewModel) {
        this.viewModel = viewModel;
    }
}

File: lib_base/src/main/java/com/czl/lib_base/mvvm/viewmodel/ItemViewModel.kt
package com.czl.lib_base.mvvm.viewmodel

import com.czl.lib_base.base.BaseViewModel
import com.czl.lib_base.data.DataRepository
import me.goldze.mvvmhabit.base.BaseModel

open class ItemViewModel<VM : BaseViewModel<*>?>(protected var viewModel: VM)